packages/core-container/__tests__/__stubs__/plugin-a.js

```diff
@@ -9,7 +9,9 @@ exports.plugin = {
   register: (container, options) => {
     return {
       container,
-      options
+      options,
+      deregister: deregisterMockFn
```

Comment:
Why is necessary adding the `deregisterMockFn` in 2 different places?

Reply:
Because of how the plugins are registered in the code. The obj that ends up in the Awilix container, is not the same one that the plugins export. The 'deregister' property needs to be set on the exported object. However, you cannot access this object from the Awilix container.

This is the approach I chose such that I can still acquire a reference to the mock function(from Awilix), while still ensuring that it's called by the PluginRegistrar plumbing

Reply:
The `deregister` method should be on the root level only. Other packages, such as `core-database-sequelize` and `core-p2p`, are using it that way.

Maybe the problem is other thing, such as https://github.com/ArkEcosystem/core/blob/2c02ef64ac5d2f90fc47858ba4e5a71144167be1/packages/core-container/lib/registrars/plugin.js#L63.

Reply:
So @j-a-m-l , given the way the plugins are registered right now(file based approach) this is one approach. 
A second approach is to have the test manually 'require()' all the plugins that are supposed to have deregister to get a reference to the mockFn; that will remove the need for 2 references. I'll make those changes